lstchain/calib/camera/calibration_calculator.py

```diff
@@ -0,0 +1,258 @@
+    """
+    Calibration calculator for LST camera
+    Fills the MonitoringCameraContainer on the base of calibration events
+
+    Parameters:
+    ----------
+    minimum_hg_charge_median :
+              Temporary cut on HG charge till the calibox TIB do not work
+             (default for filter 5.2)
+
+    maximum_lg_charge_std
+             Temporary cut on LG std against Lidar events till the calibox TIB do not work
+            (default for filter 5.2)
+
+    time_calibration_path:
+            Path with the drs4 time calibration corrections
+    """
+
+    minimum_hg_charge_median = Float(
+        5000,
+        help='Temporary cut on HG charge till the calibox TIB do not work (default for filter 5.2)'
+    ).tag(config=True)
+
+    maximum_lg_charge_std = Float(
+        300,
+        help='Temporary cut on LG std against Lidar events till the calibox TIB do not work (default for filter 5.2) '
+    ).tag(config=True)
+
+    def __init__(self, **kwargs):
+        """
+         Calibration calculator for LST camera
+         Fills the MonitoringCameraContainer on the base of calibration events
+
+         Parameters:
+         ----------
+         minimum_hg_charge_median :
+             Temporary cut on HG charge till the calibox TIB do not work
+             (default for filter 5.2)
+
+         maximum_lg_charge_std
+             Temporary cut on LG std against Lidar events till the calibox TIB do not work
+            (default for filter 5.2)
+
+        """
+        super().__init__(**kwargs)
+
+    def calculate_calibration_coefficients(self, event):
+        """
+        Calculate calibration coefficients from flatfield and pedestal statistics
+        associated to the present event
+
+        Parameters
+        ----------
+        event: EventAndMonDataContainer
+
+        """
+
+        ped_data = event.mon.tel[self.tel_id].pedestal
+        ff_data = event.mon.tel[self.tel_id].flatfield
+        status_data = event.mon.tel[self.tel_id].pixel_status
+        calib_data = event.mon.tel[self.tel_id].calibration
+
+        # mask from pedestal and flat-field data
+        monitoring_unusable_pixels = np.logical_or(status_data.pedestal_failing_pixels,
+                                                   status_data.flatfield_failing_pixels)
+        # calibration unusable pixels are an OR of all masks
+        calib_data.unusable_pixels = np.logical_or(monitoring_unusable_pixels,
+                                                   status_data.hardware_failing_pixels)
+        # Extract calibration coefficients with F-factor method
+        # Assume fix F2 factor, F2=1+Var(gain)/Mean(Gain)**2 must be known from elsewhere
+        F2 = 1.2
+
+        # calculate photon-electrons
+        n_pe = F2 * (ff_data.charge_median - ped_data.charge_median) ** 2 / (
+                ff_data.charge_std ** 2 - ped_data.charge_std ** 2)
+
+        # fill WaveformCalibrationContainer (this is an example)
```

Comment:
What do you mean with "this is an example"?

Reply:
historical comment ;-)